### PR TITLE
312 track swaps + xcp fee estimates

### PR DESCRIPTION
**Pull request by dime-squanderer**

### CODE REVIEW

File: lib/data/sources/network/api/v2_api.dart [hunk 1]
```diff
@@ -4613,6 +4613,9 @@ abstract class V2Api {
     @Query("event_name") String? eventName,
   ]);
 
+  @GET("/compose/attach/estimatexcpfees")
```
[the remark this answers is not in the document]
Comment (dime-squanderer) [reply lib/data/sources/network/api/v2_api.dart:4616]: attaches can be associated with an xcp fee. we need to make the user aware of this fee, and also make sure the user has enough xcp to cover the fee. sometimes the fee is 0.

File: lib/data/sources/network/mempool_space_client.dart [hunk 2]
```diff
@@ -45,4 +45,171 @@ class MempoolSpaceApi {
...
+      Network.regtest => throw UnsupportedError(
+          'MempoolSpace.getTransactionInfo not supported on regtest network.')
+    };
+
+    final response = await _dio.get(url);
+    return MempoolSpaceTransaction.fromJson(response.data);
+  }
+}
+
+class MempoolSpaceTransaction {
```
[the remark this answers is not in the document]
Comment (dime-squanderer) [reply lib/data/sources/network/mempool_space_client.dart:62]: mempool transactions are the only decoded txs that have all the available information to determine a swap.

File: lib/data/sources/repositories/events_repository_impl.dart [hunk 3]
```diff
@@ -79,6 +86,39 @@ class VerboseEventMapper {
...
+        final transactionInfo =
+            await transactionInfoUseCase.call(apiEvent.txHash!);
+
+        // atomic swaps will have at least 2 different input sources
+        final isAtomicSwap = _isAtomicSwap(transactionInfo);
+
+        // the btc that was swapped for the asset is held in the vout of the _other_ holder's address
+        // the output with value 546 and 547 are the values for attaching utxos, so we exclude them
+        if (isAtomicSwap) {
+          final bitcoinSwapOutputs = transactionInfo.outputs
```
[the remark this answers is not in the document]
Comment (dime-squanderer) [reply lib/data/sources/repositories/events_repository_impl.dart:104]: yeah I can move this out to the helper

File: lib/data/sources/repositories/transaction_info_repository_mempool_space_impl.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import "package:fpdart/fpdart.dart";
+import 'package:horizon/data/sources/network/mempool_space_client.dart';
```
[the remark this answers is not in the document]
Comment (dime-squanderer) [reply lib/data/sources/repositories/transaction_info_repository_mempool_space_impl.dart:2]: esplora doesn't provide the address of the inputs and outputs

**Review by mattmarcello — APPROVED** [review]

**Pull request merged**